src/core/ipv4/ip4.c

```diff
@@ -320,6 +320,120 @@ ip4_canforward(struct pbuf *p)
   return 1;
 }
 
+#if IP_NAPT
```

Comment:
Can we make it available only if we allow Tx on the the same netif?
```suggestion
#if IP_NAPT && IP_FORWARD_ALLOW_TX_ON_RX_NETIF
```